src/hooks/Filters/product-context.jsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useReducer,useContext } from "react";
+import { useProducts } from "../useProduct";
+import { filterProducts } from "./categoryFilter";
+import { discountedItems } from "./discountPercentage-filter";
+import { pricedProducts } from "./priceFilter";
+import { reducerFunction } from "./reducerFunction";
+import { sortedProducts } from "./sort_priceFilter";
+import { starredProducts } from "./starredProduct";
+const ProudctContext = createContext();
+export const useSortedProduct=()=>useContext(ProudctContext)
+export function ProductProvider({ children }) {
+    const [state, dispatch] = useReducer(reducerFunction,
+        {
```

Comment:
You can extract line 13 to line 21 out into a variable and pass that variable here.
Just for clean code. Otherwise it looks fine

Reply:
done.